src/comment_worker.py

```diff
@@ -479,9 +482,14 @@ def firm(self, sess, comment, investor, firm_name=None):
         else:
             flair_role = "Floor Trader"
 
+        # Assigns the correct firm for the flair update below
+        flair_firm = sess.query(Firm).\
+            filter(Firm.id == investor.firm).\
+            first()
+
         if not config.TEST:
             for subreddit in config.SUBREDDITS:
-                REDDIT.subreddit(subreddit).flair.set(investor.name, f"{firm.name} | {flair_role}")
+                REDDIT.subreddit(subreddit).flair.set(investor.name, f"{flair_firm.name} | {flair_role}")
```

Comment:
This is a very good fix! Approved for this one.

Reply:
Thanks! It's actually one of the first fixes I wrote, before the roles, but then I decided to package it all together instead of recommending one fix at a time.